mrp_production_request/models/stock_rule.py

```diff
@@ -116,7 +116,7 @@ def _run_production_request(
     def _run_manufacture(self, procurements):
         for procurement, _rule in procurements:
             if self._need_production_request(procurement.product_id):
-                return self._run_production_request(
+                self._run_production_request(
```

Comment:
You need to pop the procurements you already covered with production requests, otherwise after both an MR and MO will be created.

Reply:
@LoisRForgeFlow 

Sorry Lois but what do you mean? I have created this PR because I have different products under the procurements. 
For example
procurements[0] : product 1 
procurements[1] : product 2

Then I  remove it to have one more MR for each product under the procurements. 

Could you  please change the code  your suggestion? I did not get or  are you not approving this new  approach?

Thank you so much in advance.

Reply:
This iteration is resolving procurements os some product in a different way than Odoo standard (using manufacturing requests), if you after that call super with the exact same set of procurements you will also create MO's for the procurements you have just created MR's in that iteration.

you need to pop (extract from the set of `procurements`) the ones that have already created a MR and then call super with a reduced set of procurements:

`return super()._run_manufacture(procurements_not_resolved_with_manufacturing_request)`

Reply:
Hi @LoisRForgeFlow 

Commit created for this

Reply:
Hi Lois 

New commit pushed

Reply:
Hi @LoisRForgeFlow  Could u please review it ?

Reply:
Hey @LoisRForgeFlow 

Could u pleae review it ?